### PR TITLE
[1249] Latent diffusion forecast engine target

### LINKED ISSUE #1221: Integrate diffusion model into trainer
### Describe the task. It can be a feature, documentation, etc.

Once the diffusion model has been implemented (#1219) and the preprocessing functionality is ready (#1220), the diffusion model can be integrated into the trainer class.

It still needs to be decided how the forecast engine type can be specified. I.e., whether to train a latent diffusion model or a deterministic forecast engine end-to-end.

### Hedgedoc URL, if you are keeping notes, plots, logs in hedgedoc.

_No response_

### Area

- [ ] datasets, data readers, data preparation and transfer
- [x] model
- [ ] science
- [ ] infrastructure and engineering
- [ ] evaluation, export and visualization
- [ ] documentation

### LINKED ISSUE #1249: Targets for latent diffusion model training
### Describe the task. It can be a feature, documentation, etc.

To train a latent diffusion forecast engine, we need to encode targets into the latent space using a pre-trained encoder.

This issue deals with preparing the code such that a latent target can be obtained. A similar approach was implemented in #1019.

Coordinate with the abstract loss class implementation #1210 and align with student-teacher implementation #1111 for a later "seamless" integration.

### Hedgedoc URL, if you are keeping notes, plots, logs in hedgedoc.

_No response_

### Area

- [ ] datasets, data readers, data preparation and transfer
- [x] model
- [ ] science
- [ ] infrastructure and engineering
- [ ] evaluation, export and visualization
- [ ] documentation

### PR DESCRIPTION
**Pull request by MatKbauer**
## Description

This PR uses the target_and_aux_calculator to encode the target into latent state for the loss calculation of the latent diffusion forecast engine.

Along these lines, the encoding steps are subsumed into a single `model.encode()` function. This function is used to encode the targets without gradients.

## Issue Number

Ref #1268 (encapsulate encoding steps into a single function)
Closes #1249 (target encoding for latent diffusion forecast engine)
Closes #1221 (integrate diffusion model into trainer)

## Checklist before asking for review

-   [x] I have performed a self-review of my code
-   [x] My changes comply with basic sanity checks:
      - I have fixed formatting issues with `./scripts/actions.sh lint`
      - I have run unit tests with `./scripts/actions.sh unit-test`
      - I have documented my code and I have updated the docstrings.
      - I have added unit tests, if relevant
-   [x] I have tried my changes with data and code:
      - I have run the integration tests with `./scripts/actions.sh integration-test`
      - (bigger changes) I have run a full training and I have written in the comment the run_id(s): `launch-slurm.py --time 60`
      - (bigger changes and experiments) I have shared a hegdedoc in the github issue with all the configurations and runs for this experiments
-   [ ] I have informed and aligned with people impacted by my change:
    - for config changes: the MatterMost channels and/or a design doc
    - for changes of dependencies: the MatterMost software development channel

### CODE REVIEW

**Comment by tjhunter** [comment]
@MatKbauer  do we still need this PR? I presume it has been combined with other PRs.

**Comment by MatKbauer** [comment]
Completed and part of #1300
